pkg/api/validate.go

```diff
@@ -0,0 +1,60 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package api
+
+import fmt "fmt"
+
+func (v *ValidateContainerAdjustmentRequest) AddPlugin(name, index string) {
+	if v == nil {
+		return
+	}
+
+	v.Plugins = append(v.Plugins, &PluginInstance{
+		Name:  name,
+		Index: index,
+	})
+}
+
+func (v *ValidateContainerAdjustmentRequest) AddResponse(rpl *CreateContainerResponse) {
+	if v == nil {
+		return
+	}
+
+	v.Adjust = rpl.Adjust
+	v.Update = rpl.Update
+}
+
+func (v *ValidateContainerAdjustmentRequest) AddOwners(owners *OwningPlugins) {
+	if v == nil {
+		return
+	}
+
+	v.Owners = owners
+}
+
+func (v *ValidateContainerAdjustmentResponse) ValidationResult(plugin string) error {
+	if v == nil || !v.Reject {
+		return nil
+	}
+
+	reason := v.Reason
+	if reason == "" {
+		reason = "unknown rejection reason"
+	}
+
+	return fmt.Errorf("validator %q rejected container creation, reason: %s", plugin, reason)
```

Comment:
This phrasing assumes when it will be called, will "creation" always be accurate? What if we adjusted this to "rejected container adjustment"?

Reply:
I'll rephrase it like that.

Reply:
Updated error message accordingly.